### PR TITLE
Add Kafka 3.1.0 and remove 2.8.x

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lukywill16@gmail.com>

This PR changes several things:
- adds Kafka 3.1.0 and set it as default version
- remove Kafka 2.8.x versions
- increase version of Strimzi OAuth

### CODE REVIEW

File: examples/http/http-producer.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ spec:
               value: "100"
             - name: LOG_LEVEL
               value: DEBUG
-          image: quay.io/strimzi-test-clients/test-client-http-producer:latest-kafka-3.0.0
+          image: quay.io/strimzi-test-clients/test-client-http-producer:latest
```
Comment (sknot-rh) [inline examples/http/http-producer.yaml:35]: Why is `latest` used there?

Comment (im-konge) [reply examples/http/http-producer.yaml:35]: On the `main`, we are using the `latest` tag. For releases it is changed to the release tag - for example here: https://github.com/strimzi/test-clients/blob/0.1.1/examples/kafka/kafka-consumer.yaml

Comment (sknot-rh) [reply examples/http/http-producer.yaml:35]: Thanks

**Review by Frawless — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -10,19 +10,25 @@
...
         <module>kafka</module>
         <module>http</module>
     </modules>
 
+    <repositories>
+        <repository>
+            <id>kafka-staging</id>
+            <url>https://repository.apache.org/content/groups/staging/</url>
+        </repository>
+    </repositories>
```
Comment (scholzj) [inline pom.xml:22]: You shouůd remove this.

**Review by scholzj — APPROVED** [review]

**Pull request merged**